### PR TITLE
[SPARK-35600][TESTS] Move Set command related test cases to SetCommandSuite

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

Move `Set` command related test cases from `SQLQuerySuite` to a new test suite `SetCommandSuite`. There are 7 test cases in total.

### Why are the changes needed?

Code refactoring. `SQLQuerySuite` is becoming big. 

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Unit tests

### CODE REVIEW

**Review by cloud-fan — APPROVED** [review]

File: sql/core/src/test/scala/org/apache/spark/sql/SetSuite.scala [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql
+
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.test.{SharedSparkSession, TestSQLContext}
+import org.apache.spark.util.ResetSystemProperties
+
+class SetSuite extends QueryTest with SharedSparkSession with ResetSystemProperties  {
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/SetSuite.scala:24]: Maybe, `SetSuite` -> `SetCommandSuite`?

**Comment by dongjoon-hyun** [comment]
Retest this please

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (one minor comment about the test suite naming)

File: sql/core/src/test/scala/org/apache/spark/sql/SetSuite.scala (hunk 1 above)
Comment (gengliangwang) [reply sql/core/src/test/scala/org/apache/spark/sql/SetSuite.scala]: Done

**Comment by dongjoon-hyun** [comment]
Thank you for update, @gengliangwang .

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**